server/visualdl/lib.py

```diff
@@ -41,13 +45,23 @@ def get_scalar(storage, mode, tag, num_records=300):
         if len(data) <= num_records:
             return data
 
-        span = float(len(data) / num_records)
-        end_idx = len(data) - 1
-        res = []
-        for i in xrange(num_records):
-            id = int(end_idx - i * span)
-            res.append(data[id])
-        return [v for v in reversed(res)]
+        # if sample amount is larger than actual data amount, then choose all data
+        if num_records > len(data):
+            num_records = len(data)
+
+        span = float(len(data)) / (num_records - 1)
+        span_multiply = 0
+
+        data_idx = int(span_multiply * span)
+        sampled_data = []
+
+        while data_idx < len(data):
+            sampled_data.append(data[data_idx])
+            span_multiply += 1
+            data_idx = int(span_multiply * span)
+
+        sampled_data.append(data[-1])
```

Comment:
sample from data[-1] to data[0], so the latest record can be taken naturally.

Reply:
done